source/statements.tex

```diff
@@ -810,14 +810,17 @@
 \indextext{\idxcode{return}!constructor and}%
 \indextext{\idxcode{return}!constructor and}%
 A \tcode{return} statement with an operand of type \keyword{void} shall be used only
-in a function whose return type is \cv{}~\keyword{void}.
+in a function that has a \cv{}~\keyword{void} return type.
 A \tcode{return} statement with any other operand shall be used only
-in a function whose return type is not \cv{}~\keyword{void};
+in a function that has a return type other than \cv{}~\keyword{void};
 \indextext{conversion!return type}%
 the \tcode{return} statement initializes the
 glvalue result or prvalue result object of the (explicit or implicit) function call
 by copy-initialization\iref{dcl.init} from the operand.
 \begin{note}
+A constructor or destructor does not have a return type.
```

Comment:
If we claim that the constructor of a class does not have a return type, consider [dcl.init.general] p15
>  if the function is a constructor, the call is a prvalue of the cv-unqualified version of the destination type whose result object is initialized by the constructor.  

> A function call is an lvalue if the result type is an lvalue reference type or an rvalue reference to function type, an xvalue if the result type is an rvalue reference to object type, and a prvalue otherwise.  

These rules will be conflict. How could the call of a constructor be a prvalue of the **cv-unqualified version of the destination type**?

Reply:
@jensmaurer Final thoughts?

Reply:
Constructors / destructors don't have a return type in the "function declaration" sense, and that absence makes "return something" syntactically ill-formed.  I think the quoted stuff doesn't change that; it just means that a constructor call (in the context of copy-initialization, not in general) has (is considered to have) a certain type and value category. That has no bearing on the syntactic constraints on constructor declarations.

I thinks this should go in.